xds/src/main/java/io/grpc/xds/ClusterImplLoadBalancer.java

```diff
@@ -129,7 +132,12 @@ public boolean acceptResolvedAddresses(ResolvedAddresses resolvedAddresses) {
     childLbHelper.updateDropPolicies(config.dropCategories);
     childLbHelper.updateMaxConcurrentRequests(config.maxConcurrentRequests);
     childLbHelper.updateSslContextProviderSupplier(config.tlsContext);
-    childLb.handleResolvedAddresses(
+
+    if (!config.childPolicy.getProvider().getPolicyName().equals(childPolicy)) {
+      childSwitchLb.switchTo(config.childPolicy.getProvider());
```

Comment:
Always switch. It handles this optimization (using provider instances)

Reply:
Got the idea from 'PriorityLoadBalancer` that does this check. Removed now.